src/box/lua/trigger.c

```diff
@@ -0,0 +1,242 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include "box/lua/func_adapter.h"
+#include "core/event.h"
+#include <diag.h>
+#include <fiber.h>
+#include "lua/utils.h"
+
+/**
+ * Sets a trigger with passed name to the passed event.
+ * The first argument is event name, the second one is trigger name, the third
+ * one is new trigger handler - it can be a Lua function or another callable
+ * object. If there is an already registered trigger with such name in the
+ * event, it is replaced with new trigger.
+ * Returns new trigger handler (which was passed as the third argument).
+ */
+static int
+luaT_trigger_set(struct lua_State *L)
+{
+	const char *event_name = luaL_checkstring(L, 1);
+	const char *trigger_name = luaL_checkstring(L, 2);
+	if (lua_gettop(L) < 3 || !luaL_iscallable(L, 3))
+		luaL_typerror(L, 3, "callable");
+	/*
+	 * The following code is written in assumption no error will be thrown.
+	 */
+	struct event *event = event_get(event_name, true);
+	assert(event != NULL);
+	struct func_adapter *func = func_adapter_lua_create(L, 3);
+	event_reset_trigger(event, trigger_name, func);
+	func_adapter_lua_get_func(func, L);
```

Comment:
Nit: This line doesn't seem to be necessary because the function is already at the top of the stack.

Reply:
Actually, no. I didn't check number of arguments, so if I call `trigger.set('event', 'trigger', handler, 'redundant')`, `'redundant'` will be at the top. So I added the check to all 4 public methods of the module, covered it with the test and removed this redundant `func_adapter_lua_get_func`.